modules/ContextUtils.js

```diff
@@ -0,0 +1,114 @@
+import React, { PropTypes } from 'react'
+
+// Works around issues with context updates failing to propagate.
+// https://github.com/facebook/react/issues/2517
+// https://github.com/reactjs/react-router/issues/470
+
+export function createContextProvider(name, contextType = PropTypes.any) {
+  const ContextProvider = React.createClass({
+    propTypes: {
+      children: PropTypes.node.isRequired
+    },
+
+    contextTypes: {
+      [name]: contextType
+    },
+
+    childContextTypes: {
+      [name]: contextType
+    },
+
+    getChildContext() {
+      return {
+        [name]: {
+          ...this.context[name],
+          subscribe: this.subscribe,
+          eventIndex: this.eventIndex
+        }
+      }
+    },
+
+    componentWillMount() {
+      this.eventIndex = 0
+      this.listeners = []
+    },
+
+    componentWillReceiveProps() {
+      this.eventIndex++
+    },
+
+    componentDidUpdate() {
+      this.listeners.forEach(listener => listener(this.eventIndex))
+    },
+
+    subscribe(listener) {
+      // No need to immediately call listener here.
+      this.listeners.push(listener)
+
+      return () => {
+        this.listeners = this.listeners.filter(item => item !== listener)
+      }
+    },
+
+    render() {
+      return this.props.children
+    }
+  })
+
+  return ContextProvider
+}
+
+export function connectToContext(WrappedComponent, name, contextType = PropTypes.any) {
```

Comment:
I agree with @ryanflorence here. This doesn't feel like the right place for all of this. I know you want to build out a separate library, but I don't think it's the right place to live-test this approach. We already have a `listen` and `router` is on context, so this can probably be simplified quite a bit using existing code.